libs/template/experimental/virtual-scrolling/src/lib/scroll-strategies/fixed-size-virtual-scroll-strategy.ts

```diff
@@ -0,0 +1,290 @@
+import {
+  Directive,
+  EmbeddedViewRef,
+  inject,
+  Input,
+  NgIterable,
+  OnChanges,
+  OnDestroy,
+  SimpleChanges,
+} from '@angular/core';
+import { coalesceWith } from '@rx-angular/cdk/coalescing';
+import {
+  combineLatest,
+  MonoTypeOperatorFunction,
+  ReplaySubject,
+  Subject,
+} from 'rxjs';
+import {
+  distinctUntilChanged,
+  map,
+  startWith,
+  switchMap,
+  takeUntil,
+  tap,
+} from 'rxjs/operators';
+
+import {
+  ListRange,
+  RxVirtualForViewContext,
+  RxVirtualScrollStrategy,
+  RxVirtualScrollViewport,
+  RxVirtualViewRepeater,
+} from '../model';
+import { unpatchedAnimationFrameTick } from '../util';
+import {
+  DEFAULT_ITEM_SIZE,
+  DEFAULT_RUNWAY_ITEMS,
+  DEFAULT_RUNWAY_ITEMS_OPPOSITE,
+  RX_VIRTUAL_SCROLL_DEFAULT_OPTIONS,
+} from '../virtual-scroll.config';
+
+/**
+ * @Directive FixedSizeVirtualScrollStrategy
+ *
+ * @description
+ *
+ * The `FixedSizeVirtualScrollStrategy` provides a very performant way of rendering
+ * items of a given size. It is comparable to \@angular/cdk `FixedSizeVirtualScrollStrategy`, but
+ * with a high performant layouting technique.
+ *
+ * @docsCategory RxVirtualFor
+ * @docsPage RxVirtualFor
+ * @publicApi
+ */
+@Directive({
+  // eslint-disable-next-line @angular-eslint/directive-selector
+  selector: 'rx-virtual-scroll-viewport[itemSize]',
+  providers: [
+    {
+      provide: RxVirtualScrollStrategy,
+      useExisting: FixedSizeVirtualScrollStrategy,
+    },
+  ],
+  standalone: true,
+})
+// eslint-disable-next-line @angular-eslint/directive-class-suffix
+export class FixedSizeVirtualScrollStrategy<
```

Comment:
It looks like a lot of code is duplecated between strategies, would it makes sense to factorise?

Reply:
autosize & dynamic size share a lot of code. would be nice to factorize, yes